src/main/kotlin/biokotlin/cli/MutateProteins.kt

```diff
@@ -0,0 +1,380 @@
+ */
+class MutateProteins : CliktCommand(help = "Mutate Proteins") {
+
+    private val myLogger = LogManager.getLogger(MutateProteins::class.java)
+
+    enum class TypeMutation {
+        DELETION, POINT_MUTATION, INSERTION
+    }
+
+    private val proteinLetters = arrayOf(
+        'R', 'H', 'K', 'D', 'E', 'S', 'T', 'N', 'Q', 'C', 'G', 'P', 'A', 'V', 'I', 'L', 'M', 'F', 'Y', 'W'
+    )
+
+    private val numProteinLetters = proteinLetters.size
+
+    val inputFasta by option(help = "Full path to input fasta file")
+        .required()
+
+    val outputFasta by option(help = "Full path to output fasta file")
+        .required()
+
+    val bedfile by option(help = "Full path to bedfile")
+        .required()
+
+    val typeMutation by option(help = "Type of mutation")
+        .enum<TypeMutation>()
+        .default(TypeMutation.POINT_MUTATION)
+
+    val putMutationsInRanges by option(help = "Put mutations in ranges defined by bedfile")
+        .boolean()
+        .default(true)
+
+    val length by option(help = "Length of the deletion mutation")
+        .int()
+        .default(5)
+
+    val numMutations by option(help = "Number of point mutations")
+        .int()
+        .default(10)
+
+    val mutatedIndicesBedfile by option(help = "Full path to bedfile to output mutated indices")
+
+    val randomSeed by option(help = "Random seed")
+        .int()
+        .default(1234)
+
+    override fun run() {
+
+        val ranges = readBedfile()
+
+        val reader = FastaIO(inputFasta, SeqType.protein)
+
+        val mutatedIndicesWriter = if (mutatedIndicesBedfile != null) bufferedWriter(mutatedIndicesBedfile!!) else null
+
+        bufferedWriter(outputFasta).use { writer ->
+
+            val ids = mutableSetOf<String>()
+            var count = 0
+            while (reader.hasNext()) {
+
+                val record = reader.next() as ProteinSeqRecord
+                val id = record.id
+                ids.add(id)
+                count++
+
+                val mutatedSeq = when (typeMutation) {
+                    TypeMutation.DELETION -> deletionMutation(record, ranges)
+                    TypeMutation.POINT_MUTATION -> pointMutations(record, ranges, mutatedIndicesWriter)
+                    TypeMutation.INSERTION -> insertionMutation(record, ranges)
+                }
+
+                writeSeq(writer, id, mutatedSeq)
+
+            }
+
+            mutatedIndicesWriter?.close()
+
+            if (count != ids.size) {
+                myLogger.warn("Duplicate IDs found")
+            }
+
+        }
+
+    }
+
+    /**
+     * Write sequence to writer. The sequence is written in fasta format.
+     */
+    private fun writeSeq(writer: BufferedWriter, id: String, mutatedSeq: String) {
+        writer.write(">${id}\n")
+        mutatedSeq
+            .chunked(60)
+            .forEach { line ->
+                writer.write(line)
+                writer.newLine()
+            }
+    }
+
+    // zero based inclusive / inclusive
```

Comment:
You are converting this to be 0 based inclusive inclusive?  Consider using a different name because Bed Regions are 0 based inclusive exclusive.  This parsing of BED files and the writeSeq above should be Biokotlin utilities if they don't already exist(we have new Fasta IO thanks to Alagu's new commits).

Reply:
I suggest making this another task after I've refactored the BioKotlin repository to remove Clikt?